knitout-to-kcode.js

```diff
@@ -156,28 +157,34 @@ function merge_types(a,b) {
 		if (a !== TYPE_XFER_FOUR_PASS && a !== TYPE_XFER_TWO_PASS) return a;
 		else return null;
 	}
-	
+
 	//types that only define one bed get merged:
 	if (a === TYPE_KNIT_x) {
-		if      (b === TYPE_x_KNIT || b === TYPE_KNIT_KNIT) return TYPE_KNIT_KNIT;
+		if (b === TYPE_x_KNIT || b === TYPE_KNIT_KNIT) return TYPE_KNIT_KNIT;
 		else if (b === TYPE_x_TUCK || b === TYPE_KNIT_TUCK) return TYPE_KNIT_TUCK;
+		else if (b === TYPE_SPLIT_TO_BACK) return TYPE_KNITFRONT_SPLIT2BACK; //* //?
 	} else if (a === TYPE_x_KNIT) {
-		if      (b === TYPE_KNIT_x || b === TYPE_KNIT_KNIT) return TYPE_KNIT_KNIT;
+		if (b === TYPE_KNIT_x || b === TYPE_KNIT_KNIT) return TYPE_KNIT_KNIT;
 		else if (b === TYPE_TUCK_x || b === TYPE_TUCK_KNIT) return TYPE_TUCK_KNIT;
+		else if (b === TYPE_SPLIT_TO_FRONT) return TYPE_KNITBACK_SPLIT2FRONT; //* //?
 	} else if (a === TYPE_TUCK_x) {
 		if      (b === TYPE_x_KNIT || b === TYPE_TUCK_KNIT) return TYPE_TUCK_KNIT;
 		else if (b === TYPE_x_TUCK || b === TYPE_TUCK_TUCK) return TYPE_TUCK_TUCK;
 	} else if (a === TYPE_x_TUCK) {
 		if      (b === TYPE_KNIT_x || b === TYPE_KNIT_TUCK) return TYPE_KNIT_TUCK;
 		else if (b === TYPE_TUCK_x || b === TYPE_TUCK_TUCK) return TYPE_TUCK_TUCK;
-	} else if (a === TYPE_KNIT_KNIT) {
+	} else if (a === TYPE_KNIT_KNIT) { //* can split append to passes with knitting on both beds //?
 		if (b === TYPE_KNIT_x || b === TYPE_x_KNIT) return TYPE_KNIT_KNIT;
 	} else if (a === TYPE_KNIT_TUCK) {
 		if (b === TYPE_KNIT_x || b === TYPE_x_TUCK) return TYPE_KNIT_TUCK;
 	} else if (a === TYPE_TUCK_KNIT) {
 		if (b === TYPE_TUCK_x || b === TYPE_x_KNIT) return TYPE_TUCK_KNIT;
 	} else if (a === TYPE_TUCK_TUCK) {
 		if (b === TYPE_TUCK_x || b === TYPE_x_TUCK) return TYPE_TUCK_TUCK;
+	} else if (a === TYPE_SPLIT_TO_BACK || a === TYPE_KNITFRONT_SPLIT2BACK) {
```

Comment:
Separate `SPLIT2BACK` token is again making things more complicated but not clear it's providing any new guarantee? (Would a `SPLIT_TO_BACK` with some knits be handled any differently than one without?)